### PR TITLE
dbeaver/pro#2134 Rendering improvements

### PR DESCRIPTION
**Pull request by ShadelessFox**

### CODE REVIEW

**Review by E1izabeth — APPROVED** [review]

**Comment by serge-rider** [comment]
Not compiling

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/e4/DBeaverStackRenderer.java [hunk 1]
```diff
@@ -49,21 +54,55 @@
...
 
     private static final Log log = Log.getLog(DBeaverStackRenderer.class);
 
+    @Inject
+    @Optional
+    void subscribePerspectiveSwitched(@NotNull IEventBroker broker) {
+        final EventHandler handler = new EventHandler() {
+            @Override
+            public void handleEvent(Event event) {
+                final Object element = event.getProperty(UIEvents.EventTags.ELEMENT);
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/e4/DBeaverStackRenderer.java]: Let's protect it with try/catch throwable.
I am afraid that old Eclipse version or some other thing may cause an unexpected error and may break entire app.

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Comment by uslss** [comment]
verified

**Review by serge-rider — APPROVED** [review]

**Comment by pzygielo** [comment]
- Seems that this _improvement_ caused #22275 which was observed almost immediately, but this feature for kids is still there, enabled by default.

**Pull request merged**